examples/blas/run_time_dispatching/level3/CMakeLists.txt

```diff
@@ -0,0 +1,75 @@
+
+#===============================================================================
+# Copyright 2020-2021 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+#
+# SPDX-License-Identifier: Apache-2.0
+#===============================================================================
+
+# NOTE: user need to set env var SYCL_DEVICE_FILTER to use runtime example without specifying backend in CMake
+# $ENV{SYCL_DEVICE_FILTER}
+
+
+# Build object from all example sources
+set(BLAS_RT_SOURCES "gemm_usm")
+
+# Set up for the right backend for run-time dispatching examples
+# If users build more than one backend (i.e. mklcpu and mklgpu, or mklcpu and CUDA), they may need to
+# overwrite SYCL_DEVICE_FILTER in their environment to run on the desired backend
+set(DEVICE_FILTERS "")
+if(ENABLE_MKLCPU_BACKEND)
+  list(APPEND DEVICE_FILTERS "cpu")
+endif()
+if(ENABLE_MKLGPU_BACKEND)
+  list(APPEND DEVICE_FILTERS "gpu")
+endif()
+if(ENABLE_CUBLAS_BACKEND)
+  list(APPEND DEVICE_FILTERS "cuda:gpu")
+endif()
+if(ENABLE_ROCBLAS_BACKEND)
+  list(APPEND DEVICE_FILTERS "hip:gpu")
+endif()
+
+message(STATUS "SYCL_DEVICE_FILTER will be set to the following value(s): [${DEVICE_FILTERS}] for run-time dispatching examples")
+
+foreach(blas_rt_source ${BLAS_RT_SOURCES})
+  add_executable(example_${domain}_${blas_rt_source} ${blas_rt_source}.cpp)
+  target_compile_options(example_${domain}_${blas_rt_source} PRIVATE -DNOMINMAX)
+  target_include_directories(example_${domain}_${blas_rt_source}
+      PUBLIC ${PROJECT_SOURCE_DIR}/examples/include
+      PUBLIC ${PROJECT_SOURCE_DIR}/include
+      PUBLIC ${CMAKE_BINARY_DIR}/bin
+  )
+
+  add_dependencies(example_${domain}_${blas_rt_source} onemkl)
+
+  if (USE_ADD_SYCL_TO_TARGET_INTEGRATION)
```

Comment:
Can we add a comment when this code will be enabled?

Reply:
Added comments